enumerables.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = is_a?(Array) ? self : to_a
+    i = 0
+    # puts arr
+    while i < arr.length
+      yield arr[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = is_a?(Array) ? self : to_a
+    i = 0
+    # puts arr
+    while i < arr.length
+      yield arr[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = to_a
+    new_array = []
+    arr.each do |item|
+      new_array.push(item) if yield item
+    end
+    new_array
+  end
+
+  def my_all?
+    my_each { |item| return false if yield(item) == false }
+    true
+  end
+
+  def my_any?
+    my_each { |item| return true if yield(item) == true }
+    false
+  end
+
```

Comment:
- [ ] enumerables #my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true
- [ ] enumerables #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric
- [ ] enumerables #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer
- [ ] enumerables #my_any? when a Regex is passed as an argument returns true if any of the collection matches the Regex
- [ ] enumerables #my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ]  enumerables #my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
- [ ]  enumerables #my_any? when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern